GraphNG: update uPlot v1.5.0

leeoniya opened the pull request:
a lot of goodies here:

- `spline`, `stepped`, and `bars` path builders moved into uPlot core for easier maintenance and tighter integration. all have been refactored to use a new lower-level, more explicit, and faster API. the linear builder was extracted from the core to be equally as modular and consume the same api. all paths now run at peak performance. :rocket: 
- `bars` paths got some fixes with width calculations, you can see how broken it was previously by zooming into only a few bars. besides `widthFactor`, `bars` now have a `maxWidth` parameter, should we choose to expose it.
- the `outerJoinValues()` function was moved into a `uPlot.join()` utility function.
- log scales now handle values <= 0 without crashing.
- gradients supplied to `series.fill` have been fixed (was a bug in the opts cloning algorithm)
- `opts.gutters: {x,y}` is gone, replaced by a more flexible and explicit `opts.padding: [top, rgt, btm, lft]`. this is important for edge-to-edge sparklines without axes that still need padding to prevent cutting off of points and strokes at the edges. (see https://github.com/leeoniya/uPlot/issues/392 for visuals)
- a new option called `drawOrder: ["axes", "series"]` allows for drawing the grid over the series. might be needed for native-izing @ryantxu's https://github.com/NatelEnergy/grafana-discrete-panel

the second commit is unrelated to the update, but fixes a superfluous client-zoom after a selection end by setting `cursor.drag.setScale = false` and manually hiding the selection region in the `setSelect` hook.

i think that about covers it.

leeoniya committed:
bump uPlot 1.5.0

leeoniya committed:
prevent client-side zoom from triggering at the end of a selection

leeoniya committed:
"prettier"? more like, "uglier".

leeoniya committed:
update yarn.lock

leeoniya commented:
> TypeError: Cannot read property 'bars' of undefined

not sure what's going on with the CI, is it not picking up the new uPlot version? :thinking: 

(feel free to push commits to this branch to fix the tests here)

hugohaggmark committed:
PanelLibrary: Adds delete Api (#29741)

* PanelLibrary: Adds delete

* Chore: fixes some comments

* Chore: changes after PR comments

* Refactor: deletes from correct org

ryantxu commented on packages/grafana-ui/src/components/uPlot/config/UPlotConfigBuilder.ts:
```diff
@@ -68,9 +69,12 @@ export class UPlotConfigBuilder {
+    config.cursor = this.cursor || {};
+
+    // prevent client-side zoom from triggering at the end of a selection
+    defaultsDeep(config.cursor, { drag: { setScale: false } });
```
```suggestion
    config.cursor.drag = { setScale: false };
```

The `defaultsDeep` returns a value -- so I don't think this does what you think

ryantxu replied:
nevermind -- you are right!

ryantxu reviewed (approved):
looks good!

ryantxu committed:
update snapshot

dboslee committed:
AlertListPanel: Add options to sort by Time(asc) and Time(desc) (#29764)

This also makes all the sortOrderOptions available when using
the "changes" showOption.

Resolves: #29156

ryantxu committed:
Merge remote-tracking branch 'origin/master' into uplot-update

aknuds1 committed:
Move middleware context handler logic to service (#29605)

* middleware: Move context handler to own service

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Emil Tullsted <sakjur@users.noreply.github.com>
Co-authored-by: Will Browne <wbrowne@users.noreply.github.com>

aknuds1 committed:
Chore: Configure go-ruleguard via golangci-lint (#28419)

* Chore: Configure go-ruleguard via golangci-lint

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

wbrowne committed:
Plugins: Add support for signature manifest V2 (#29240)

* add support for signing manifest v2

* add log and fix var name

* shorten comment

* improve comment

* remove unnecessary param

* improve naming

* reformat

* rename var

* refactor test

* remove unnecessary assert

* simplify test requirements

* add more test cases

* address feedback

* revert naming

* flip tracking missing

* fix check

* Trigger Build

kylebrandt committed:
Expressions: Field names from refId (#29755)

so one can see which expression data came from when visualizing

torkelo committed:
GraphNG: Add test dashboard for null & and gaps rendering (#29769)

peterholmberg committed:
Alerting: Next gen Alerting page (#28397)

* create page and sidebar entry

* add components for query editor and definition

* split pane things

* add reducer and action

* implement split pane and update ui actions

* making things pretty

* Unify toolbar

* minor tweak to title prefix and some padding

* can create definitions

* fix default state

* add notificaion channel

* add wrappers to get correct spacing between panes

* include or exclude description

* implement query editor

* start on query result component

* update from master

* some cleanup and remove expressions touch ups

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

peterholmberg committed:
Expressions: Add placeholders to hint on input (#29773)

* math placeholder

* resample changes

aocenas committed:
Explore: Fix remounting of query row (#29771)

* Use passed data source in init

* Dummy commit

dprokop commented on packages/grafana-ui/src/components/uPlot/config/UPlotSeriesBuilder.ts:
```diff
@@ -1,9 +1,19 @@
-import { barsBuilder, smoothBuilder, stepBeforeBuilder, stepAfterBuilder } from '../paths';
 import { PlotConfigBuilder } from '../types';
 
+const pathBuilders = uPlot.paths;
```
@leeoniya seems like the Plot tests are failing because of this. Are `paths` exposed from uPlot for sure?

dprokop replied:
Is it possible this line is causing troubles https://github.com/leeoniya/uPlot/blob/a568fb12b6567567e9f5ce74a31d3628b56def5c/src/uPlot.js#L2346

leeoniya replied:
the only thing in the npm package is the `/dist` directory, and "module" in `package.json` points to the esm file in there, so:

https://github.com/leeoniya/uPlot/blob/530a46c53213bb63593451ae48d61e566185cb9e/dist/uPlot.esm.js#L3971-L3981

aocenas committed:
@grafana-runtime: Throw error if health check fails in DataSourceWithBackend (#29743)

mousedownmike committed:
Added httpMethod to webhook (#29780)

The values appear like they can be "POST" or "PUT" but there doesn't appear to be a place for that info in the documentation.

ryantxu committed:
Merge remote-tracking branch 'origin/master' into uplot-update

ryantxu committed:
skip test

ryantxu committed:
skip test

ryantxu commented:
Since other PRs depend on this, and sorting out the jest import is non-trivial -- i removed the test (renamed the file).  I could not do `.skip` since the failure is on the initial import not how the object gets mocked

![image](https://user-images.githubusercontent.com/705951/101941518-f1a52100-3b9c-11eb-895d-5eee5a728120.png)
